integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/datamap/CarbonCreateDataMapCommand.scala

```diff
@@ -49,10 +49,6 @@ case class CarbonCreateDataMapCommand(
     // so streaming table does not support create datamap
     mainTable =
       CarbonEnv.getCarbonTable(tableIdentifier.database, tableIdentifier.table)(sparkSession)
-    if (mainTable.isStreamingTable) {
-      throw new MalformedCarbonCommandException("Streaming table does not support creating datamap")
```

Comment:
Better remove only for pre-agg/timeseries, keep this check for remaining datamaps and add the datamap type also in exception message

Reply:
done